src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/charts/mosaic.ts

```diff
@@ -43,118 +21,193 @@ import {
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, metricOperationSchema]),
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, sumMetricOperationSchema]),
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, lastValueOperationSchema]),
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, percentileOperationSchema]),
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, percentileRanksOperationSchema]),
-      ]),
-      schema.oneOf([
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, differencesOperationSchema]),
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, movingAverageOperationSchema]),
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, cumulativeSumOperationSchema]),
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, counterRateOperationSchema]),
-      ]),
-      schema.oneOf([
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, staticOperationDefinitionSchema]),
-        schema.allOf([partitionStatePrimaryMetricOptionsSchema, formulaOperationDefinitionSchema]),
-      ]),
-    ]),
-    { minSize: 1 }
-  ),
-  /**
-   * Configure how to break down the metric (e.g. show one metric per term).
-   */
-  group_by: schema.arrayOf(
-    schema.maybe(
-      schema.oneOf([
-        schema.allOf([partitionStateBreakdownByOptionsSchema, bucketDateHistogramOperationSchema]),
-        schema.allOf([partitionStateBreakdownByOptionsSchema, bucketTermsOperationSchema]),
-        schema.allOf([partitionStateBreakdownByOptionsSchema, bucketHistogramOperationSchema]),
-        schema.allOf([partitionStateBreakdownByOptionsSchema, bucketRangesOperationSchema]),
-        schema.allOf([partitionStateBreakdownByOptionsSchema, bucketFiltersOperationSchema]),
-      ])
+function validateMosaicGroupings(obj: {
+  outer_grouping: Array<{ collapse_by?: string }>;
+  inner_grouping?: Array<{ collapse_by?: string }>;
+}) {
+  if (obj.outer_grouping.filter((def) => def.collapse_by == null).length > 1) {
+    return 'In outer grouping, only a single non-collapsed dimension is allowed when using multiple dimensions.';
+  }
+  if ((obj.inner_grouping?.filter((def) => def.collapse_by == null).length ?? 0) > 1) {
+    return 'In inner grouping, only a single non-collapsed dimension is allowed when using multiple dimensions.';
+  }
+}
+
+export const mosaicStateSchemaNoESQL = schema.object(
+  {
+    type: schema.literal('mosaic'),
+    ...sharedPanelInfoSchema,
+    ...layerSettingsSchema,
+    ...datasetSchema,
+    ...mosaicStateSharedSchema,
+    /**
+     * Primary value configuration, must define operation. Supports field-based operations (count, unique count, metrics, sum, last value, percentile, percentile ranks), reference-based operations (differences, moving average, cumulative sum, counter rate), and formula-like operations (static value, formula).
+     */
+    metric: mergeAllMetricsWithChartDimensionSchemaWithRefBasedOps(
```

Comment:
Mosaic supports only single metric, so name has changed in this PR to make it clearer